src/ol/interaction/extentinteraction.js

```diff
@@ -0,0 +1,429 @@
+goog.provide('ol.interaction.ExtentEvent');
+goog.provide('ol.interaction.Extent');
+
+goog.require('ol');
+goog.require('ol.Feature');
+goog.require('ol.MapBrowserEvent.EventType');
+goog.require('ol.MapBrowserPointerEvent');
+goog.require('ol.coordinate');
+goog.require('ol.events');
+goog.require('ol.events.Event');
+goog.require('ol.geom.GeometryLayout');
+goog.require('ol.geom.GeometryType');
+goog.require('ol.geom.Point');
+goog.require('ol.geom.Polygon');
+goog.require('ol.interaction.Interaction');
+goog.require('ol.interaction.Pointer');
+goog.require('ol.layer.Vector');
+goog.require('ol.proj');
+goog.require('ol.source.Vector');
+
+/**
+ * @classdesc
+ * Events emitted by {@link ol.interaction.ExtentInteraction} instances are instances of
+ * this type.
+ *
+ * @constructor
+ * @extends {ol.events.Event}
+ */
+ol.interaction.ExtentEvent = function() {
+  goog.base(this, 'bbox');
+};
+goog.inherits(ol.interaction.ExtentEvent, ol.events.Event);
+
+/**
+ * @classdesc
+ * Allows the user to draw a vector box by clicking and dragging on the map.
+ * Once drawn, the vector box can be modified by draging its vertices or edges.
+ * This interaction is only supported for mouse devices.
+ *
+ * @constructor
+ * @extends {ol.interaction.Pointer}
+ * @fires ol.ExtentEvent
+ * @param {olx.interaction.ExtentOptions} opts Options.
+ * @api
+ */
+ol.interaction.Extent = function(opts) {
+  /**
+   * Extent Extent
+   * @type {ol.Extent}
+   * @private
+   */
+  this.extent = null;
+  this.pointerHandler = null;
+
+  this.pixelTolerance_ = 10;
+  this.lastPixel_ = null;
+  this.snapToVertex_ = null;
+  this.snappedToVertex_ = null;
+  this.bboxFeature_ = null;
+  this.vertexFeature_ = null;
+  this.bboxOverlay_ = null;
+  this.vertexOverlay = null;
+
+  if (!opts) {
+    opts = {};
+  }
+
+  if (opts.extent) {
+    this.setExtent(opts.extent, opts.extentProjection);
+  }
+  var self = this;
+
+  /**
+   * @param {ol.MapBrowserEvent} mapBrowserEvent Event.
+   * @return {boolean} Propagate event?
+   * @this {ol.interaction.Extent}
+   * @private
+   */
+  this.handleEvent = function(mapBrowserEvent) {
```

Comment:
To match the coding style we use for other components, can you please define all member methods outside the constructor, i.e. `ol.interaction.Extent.prototype.handleEvent = function(mapBrowserEvent) ...`.